durable_storage/src/persistence_layer.rs

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+//! Implementation of the RISC-V PVM durable storage's persistence layer.
+//!
+//! A persistence layer is tied to a repository on disk, identified by a directory.
+//! Within that directory, the persistence layer needs to be able to perform basic KV operations:
+//! - get
+//! - set
+//! - delete
+//!
+//! As well as repository-level operations:
+//! - new
+//! - clone
+//! - commit (returning the commit hash)
+//! - checkout a specific commit
+
+use std::path::Path;
+
+/// Type alias for a 32-byte hash used for identifying key-value blobs & commits.
+type Hash = [u8; 32];
+
+/// Represents a key-value blob to be stored in the persistence layer.
+///
+/// Invariant: The `key` is the [`blake3`] hash of the `value`.
```

Comment:
This invariant makes this a content-addressable blob. Would you mind using to that terminology? The concept of "key" is a bit blurry, given it is mostly an implementation detail in the persistence layer.

Reply:
Not really. Could you just name it "HashedBlob" or "HashedData"? The association with keys is messy because it is ambiguous. Is it a key for the durable storage? Is it a key for the Merkle layer? You can simplify this by describing what the type is, without baking in implementation details of the Persistence layer.